### DISCUSSION TITLE
Correct way to assess predictors importance

### DISCUSSION THREAD

**Discussion by francescopisu** [ShixiangWang/self-study#82] [Q&A] [answered]
Hey, I'm wondering what do you think is the best and correct way of finding out which predictors are best for determining an outcome of interest.

Given the question is very vague, let me add some details:
I currently have a very small sample (45 patients), a number of continuous variables (20) describing heart's characteristics and a three-levels categorical variable indicating what type of condition each patient has (control, condition 1, condition 2). The dataset is balanced with 15 patients for each condition.
 
I want to determine which of these 20 variables help me most in determining the condition of interest (supposing we use the one-vs-rest approach where each time we consider the 15 patients of a condition (e.g condition 1) and the remaining 30 patients as controls).

This is my guess:
- take a look at the correlation between each independent variable (IV) and the dependent variable. 
- consider the subset of IVs with highest correlation and build an additive logistic regression model: 
`model <- glm(outcome ~ IV1 + IV2 + ... + IVn, data=data, family=binomial)`
- take a look at the model's coefficients and consider the statistically significant ones.

I'm worried about the following points:
- as far as I understand it, correlation helps me determining whether a linear relationship exist between two variables, along with its magnitude and direction. By basing my exploratory analysis on correlation, there's a risk I discard a variable that doesn't show correlation with the dependent variable because their relationship is not linear. Is it correct ?
- By building an additive model, I'm not exploring any interactions between the variables, hence losing important information. Is this correct ? 
- Given my sample is so small, I don't think I will be able to build a model with so many variables because the risk of overfitting is very high. I was reading that as a rule of thumb I should add a variable in my regression model every 10 data points.

How would you go about exploring the "Importance" of each variable in the dataset ?
Please do mind that I can use domain expertise to consider a subset of variables (from 20 to 8/10) but the problem here is the same.

Thank you very much!!

Francesco

**Comment by ShixiangWang** [comment, marked as answer]
Hi, @francescopisu This is an interesting question. I also had such thoughts when I did some modeling related work.

In this case, you have so many continuous variables but so fewer samples, I think PCA is always a good start. I will help you simplify the variables with PCs. However, this can't answer your core question, how to assess variable importance.

For exploring variable importance, I think there are many practical ways:

1. You could use stepwise model construction method (e.g., stepAIC() in R) to obtain the best logistic regression model above.
2. You can also use LASSO to cut down the variables.
3. In many common machine learning toolkits, there are usually methods to access the variable importance, e.g., for random forest, the blog https://www.r-bloggers.com/2015/06/variable-importance-plot-and-variable-selection/ may be a good reference.

I would like to respond your points above one by one.

1. Yes.
2. Yes. If you understand the background of the variables, you should think about the effects of interactions. Otherwise, I think you can ignore it at first. Because there are so many combinations for 20 variables. Let's think just one variable, the square `x` can predict `y` but `x` itself cannot (the plot shows such case). So we can expand to x^3, x^4, ... Absolutely countless. In common machine learning work, we should build models with the understanding of variables.

![image](https://user-images.githubusercontent.com/25057508/123295014-ec5f6080-d547-11eb-9c86-2e5b02881b00.png)


As for exploring correlations, my opinion is that don't just use correlation coefficients when you can visualize them.
